platforms/jvm/plugins-application/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt

```diff
@@ -251,7 +251,10 @@ DEFAULT_JVM_OPTS=${defaultJvmOpts}
 set -- \\
 <% if ( appNameSystemProperty ) {
      %>        "-D${appNameSystemProperty}=\$APP_BASE_NAME" \\
-<% } %>        -classpath "\$CLASSPATH" \\
+<% } %>\
+<% if ( !entryPointArgs.contains('-jar') ) {%>\
+    -classpath "\$CLASSPATH" \\
+<% } %>\
```

Comment:
:x: This isn't how we should be avoiding this. We should instead remove all classpath handling when it's not defined, i.e. if the `classpath` variable is empty.

Reply:
To clarify, this also includes earlier points in the script where we process the classpath.

Reply:
We're honestly not sure, it's been like that for so long the reasoning has been lost. You can remove it now, since we won't be putting that value in the script anymore.

Reply:
Thanks for the clarification, I've pushed a new commit to remove it.